File: frontend/lib/models/put/UserPutDTO.dart
class UserPutDTO {
  final String? name;
  final String? password;
  final String? email;
  final String? phone;
  final bool active;

  UserPutDTO({
    this.name,
    this.password,
    this.email,
    this.phone,
    this.active = true,
  });

  factory UserPutDTO.fromJson(Map<String, dynamic> json) {
    return UserPutDTO(
      name: json['name'],
      password: json['password'],
      email: json['email'],
      phone: json['phone'],
      active: json['active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (name != null) 'name': name,
      if (password != null) 'password': password,
      if (email != null) 'email': email,
      if (phone != null) 'phone': phone,
      'active': active,
    };
  }
}

File: backend/src/main/java/com/gymmanagement/backend/dto/put/UserPutDTO.java
package com.gymmanagement.backend.dto.put;

import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserPutDTO {

    @Size(max = 100)
    private String name;

    private String password;

    @Email
    private String email;

    @Pattern(regexp = "^[0-9]{9}$")
    private String phone;

    private boolean active = true;
}
